kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTaskTest.java

```diff
@@ -384,11 +381,36 @@ public void testVersion() {
     assertNotNull(new BigQuerySinkTask().version());
   }
 
-  // Doesn't do anything at the moment, but having this here will encourage tests to be written if
-  // the stop() method ever does anything significant
-  @Test
+  // Existing tasks should succeed upon stop is called. New tasks should be rejected once task is stopped.
+  @Test(expected = RejectedExecutionException.class)
   public void testStop() {
-    new BigQuerySinkTask().stop();
+    final String dataset = "scratch";
+    final String topic = "test_topic";
+
+    Map<String, String> properties = propertiesFactory.getProperties();
+    properties.put(BigQuerySinkConfig.TOPICS_CONFIG, topic);
+    properties.put(BigQuerySinkConfig.DATASETS_CONFIG, String.format(".*=%s", dataset));
+
+    BigQuery bigQuery = mock(BigQuery.class);
+    SinkTaskContext sinkTaskContext = mock(SinkTaskContext.class);
+    InsertAllResponse insertAllResponse = mock(InsertAllResponse.class);
+
+    when(bigQuery.insertAll(anyObject())).thenReturn(insertAllResponse);
+    when(insertAllResponse.hasErrors()).thenReturn(false);
+
+    BigQuerySinkTask testTask = new BigQuerySinkTask(bigQuery, null);
+    testTask.initialize(sinkTaskContext);
+    testTask.start(properties);
+    testTask.put(Collections.singletonList(spoofSinkRecord(topic)));
+
+    KCBQThreadPoolExecutor executor = testTask.getExecutor();
+    assertFalse((executor.isTerminating() || executor.isTerminated()));
+    testTask.stop();
+    assertTrue((executor.isTerminating() || executor.isTerminated()));
```

Comment:
Curious about this. Shouldn't isTerminating always be false and isTerminated always be true here?

Reply:
ah yeah, you are right. Forgot to change this. I wrote the test first and then added the `awaitTermination` part in `stop` :)